### PR TITLE
Added:

### PR DESCRIPTION
**Pull request by nikolasdymchenko7**
- abstract class Machine (abstract methods: doWork(), stopWork())
- classes Truck, Bulldozer, Excavator (every class overrides abstract methods from parent class Machine)
- psvm to the Main class and call methods from the Machine array in loop

### CODE REVIEW

File: src/main/java/core/basesyntax/Machine.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package core.basesyntax;
+
+public abstract class Machine {
+
```
Comment (aleksandr-bevza) [inline src/main/java/core/basesyntax/Machine.java:4]:
```suggestion
```

File: src/main/java/core/basesyntax/MainApp.java [hunk 2]
```diff
@@ -1,5 +1,13 @@
...
 
 public class MainApp {
+    public static void main(String[] args) {
+        Machine[] machinesArray = {new Truck(), new Bulldozer(), new Excavator()};
 
+        for (Machine kindOfMachine: machinesArray
+             ) {
+            kindOfMachine.doWork();
+            kindOfMachine.stopWork();
+        }
```
Comment (aleksandr-bevza) [inline src/main/java/core/basesyntax/MainApp.java:11]:
```suggestion
        for (Machine kindOfMachine: machinesArray) {
            kindOfMachine.doWork();
            kindOfMachine.stopWork();
        }
```

File: src/main/java/core/basesyntax/MainApp.java [hunk 3]
```diff
@@ -1,5 +1,13 @@
 package core.basesyntax;
 
 public class MainApp {
+    public static void main(String[] args) {
+        Machine[] machinesArray = {new Truck(), new Bulldozer(), new Excavator()};
```
Comment (aleksandr-bevza) [inline src/main/java/core/basesyntax/MainApp.java:6]:
```suggestion
```

**Review by aleksandr-bevza — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/MainApp.java (hunk 3 above)
Comment (nikolasdymchenko7) [reply src/main/java/core/basesyntax/MainApp.java]: Added fixes after review. Could you check it, please?

**Review by zachar-chorny — APPROVED** [review]

**Pull request closed without merging**